File: tools/pas2php-code-6/pas2php_include/sysutils.php
<?PHP
// 
// Converted From Pascal By Pas2Php v0.8
// Converts Object Pascal to PHP
// http://www.wascal.net/?page=pas2php
// Pas2Php v0.8 is Copyright (C) 2014-2016 Derek John Evans
// 
include_once('system.php');
include_once('phplib.php');
include_once('strutils.php');
include_once('sysutils.php');
// EmptyStr = '';
define("EmptyStr", "");
// SBoolTrue = 'True';
define("SBoolTrue", "True");
// SBoolFalse = 'False';
define("SBoolFalse", "False");
class EAbort extends Exception {
}
class EConvertError extends Exception {
}
class EAssertionFailed extends Exception {
}
class EArgumentException extends Exception {
}
class EArgumentOutOfRangeException extends EArgumentException {
}
class EAbstractError extends Exception {
}
class EFinally extends Exception {
}
// procedure Abort;
function Abort() {
  throw new EAbort("Abort");
}
// function IntToStr(const AInteger: Integer) : string;
function IntToStr($AInteger) {
  // Result : string;
  $Result = "";
  $Result = strval($AInteger);
  return $Result;
}
// function SetDirSeparators(const AString: string) : string;
function SetDirSeparators($AString) {
  // LIndex : Integer;
  $LIndex = 0;
  // Result : string;
  $Result = "";
  $Result = $AString;
  for ($LIndex = 1, $LIndex_High_ = Length($Result); $LIndex <= $LIndex_High_; $LIndex++) {
    if ($Result[($LIndex) - 1] == "\\" || $Result[($LIndex) - 1] == "/") {
      $Result[($LIndex) - 1] = DirectorySeparator;
    }
  }
  return $Result;
}
// function TryStrToInt(AString: string;out AInteger: Integer) : Boolean;
function TryStrToInt($AString, &$AInteger) {
  // Result : Boolean;
  $Result = false;
  if (Length($AString) > 0 && $AString[(1) - 1] == "\$") {
    $AString = base_convert(MidStr($AString, 2, MaxInt), 16, 10);
  }
  $AInteger = intval($AString, 10);
  $Result = strval($AInteger) == $AString;
  return $Result;
}
// function StrToInt(const AString: string) : Integer;
function StrToInt($AString) {
  // Result : Integer;
  $Result = 0;
  if (!TryStrToInt($AString, $Result)) {
    throw new EConvertError(EmptyStr);
  }
  return $Result;
}
/*
function StrToIntDef(const AString: string;const ADefault: Integer)
                     : Integer;
*/
function StrToIntDef($AString, $ADefault) {
  // Result : Integer;
  $Result = 0;
  try {
    $Result = StrToInt($AString);
  }
  catch (Exception $exception) {
    $Result = $ADefault;
  }
  return $Result;
}
/*
function TryStrToBool(const AString: string;out ABoolean: Boolean)
                      : Boolean;
*/
function TryStrToBool($AString, &$ABoolean) {
  // Result : Boolean;
  $Result = false;
  try {
    if (SameText($AString, SBoolTrue)) {
      $ABoolean = true;
    }
    else {
      if (SameText($AString, SBoolFalse)) {
        $ABoolean = false;
      }
      else {
        $ABoolean = StrToInt($AString) != 0;
      }
    }
    $Result = true;
  }
  catch (Exception $exception) {
    $Result = false;
  }
  return $Result;
}
// function StrToBool(const AString: string) : Boolean;
function StrToBool($AString) {
  // Result : Boolean;
  $Result = false;
  if (!TryStrToBool($AString, $Result)) {
    throw new EConvertError(EmptyStr);
  }
  return $Result;
}
/*
function StrToBoolDef(const AString: string;const ADefault: Boolean)
                      : Boolean;
*/
function StrToBoolDef($AString, $ADefault) {
  // Result : Boolean;
  $Result = false;
  try {
    $Result = StrToBool($AString);
  }
  catch (Exception $exception) {
    $Result = $ADefault;
  }
  return $Result;
}
/*
function BoolToStr(const ABoolean: Boolean;const ATrue: string;
                  const AFalse: string) : string;
*/
function BoolToStr($ABoolean, $ATrue = "True", $AFalse = "False") {
  // Result : string;
  $Result = "";
  if ($ABoolean) {
    $Result = $ATrue;
  }
  else {
    $Result = $AFalse;
  }
  return $Result;
}
// function FloatToStr(const AExtended: extended) : string;
function FloatToStr($AExtended) {
  // Result : string;
  $Result = "";
  $Result = strval($AExtended);
  return $Result;
}
/*
function TryStrToFloat(const AString: string;out AExtended: extended)
                       : Boolean;
*/
function TryStrToFloat($AString, &$AExtended) {
  // Result : Boolean;
  $Result = false;
  $AExtended = floatval($AString);
  $Result = strval($AExtended) == $AString;
  return $Result;
}
// function StrToFloat(const AString: string) : extended;
function StrToFloat($AString) {
  // Result : extended;
  $Result = 0.0;
  if (!TryStrToFloat($AString, $Result)) {
    throw new EConvertError(EmptyStr);
  }
  return $Result;
}
/*
function StrToFloatDef(const AString: string;const ADefault: extended)
                       : extended;
*/
function StrToFloatDef($AString, $ADefault) {
  // Result : extended;
  $Result = 0.0;
  try {
    $Result = StrToFloat($AString);
  }
  catch (Exception $exception) {
    $Result = $ADefault;
  }
  return $Result;
}
// procedure FreeAndNil(var AVariant: variant);
function FreeAndNil(&$AVariant) {
  $AVariant = Null;
}
/*
function SameText(const AString1: string;const AString2: string)
                  : Boolean;
*/
function SameText($AString1, $AString2) {
  // Result : Boolean;
  $Result = false;
  $Result = strcasecmp($AString1, $AString2) == 0;
  return $Result;
}
// function TrimLeft(const AString: string) : string;
function TrimLeft($AString) {
  // Result : string;
  $Result = "";
  $Result = ltrim($AString);
  return $Result;
}
// function TrimRight(const AString: string) : string;
function TrimRight($AString) {
  // Result : string;
  $Result = "";
  $Result = rtrim($AString);
  return $Result;
}
/*
function AnsiCompareStr(const AString1: string;const AString2: string)
                        : Integer;
*/
function AnsiCompareStr($AString1, $AString2) {
  // Result : Integer;
  $Result = 0;
  $Result = strcmp($AString1, $AString2);
  return $Result;
}
/*
function AnsiCompareText(const AString1: string;const AString2: string)
                         : Integer;
*/
function AnsiCompareText($AString1, $AString2) {
  // Result : Integer;
  $Result = 0;
  $Result = strcasecmp($AString1, $AString2);
  return $Result;
}
// function GetEnvironmentVariable(const AName: string) : string;
function GetEnvironmentVariable($AName) {
  // Result : string;
  $Result = "";
  $Result = getenv($AName);
  return $Result;
}
// function ExtractFileDir(const AFileName: string) : string;
function ExtractFileDir($AFileName) {
  // Result : string;
  $Result = "";
  $Result = dirname(SetDirSeparators($AFileName), 1);
  return $Result;
}
// function ExtractFileName(const AFileName: string) : string;
function ExtractFileName($AFileName) {
  // Result : string;
  $Result = "";
  $Result = basename(SetDirSeparators($AFileName));
  return $Result;
}
// function FileExists(const AFileName: string) : Boolean;
function FileExists($AFileName) {
  // Result : Boolean;
  $Result = false;
  $Result = is_file($AFileName);
  return $Result;
}
// function DirectoryExists(const AFileName: string) : Boolean;
function DirectoryExists($AFileName) {
  // Result : Boolean;
  $Result = false;
  $Result = is_dir($AFileName);
  return $Result;
}
// function ExtractFileExt(const AFileName: string) : string;
function ExtractFileExt($AFileName) {
  // LPos : Integer;
  $LPos = 0;
  // Result : string;
  $Result = "";
  $Result = ExtractFileName($AFileName);
  $LPos = RPos(ExtensionSeparator, $Result);
  if ($LPos == 0) {
    $Result = EmptyStr;
  }
  else {
    $Result = MidStr($Result, $LPos, MaxInt);
  }
  return $Result;
}
// function ExtractFilePath(const AFileName: string) : string;
function ExtractFilePath($AFileName) {
  // Result : string;
  $Result = "";
  $Result = ExtractFileDir($AFileName) . DirectorySeparator;
  return $Result;
}
/*
function ChangeFileExt(const AFileName: string;const AFileExt: string)
                       : string;
*/
function ChangeFileExt($AFileName, $AFileExt) {
  // LPos : Integer;
  $LPos = 0;
  // Result : string;
  $Result = "";
  $Result = ExtractFileName($AFileName);
  $LPos = RPos(ExtensionSeparator, $Result);
  if ($LPos == 0) {
    $Result = ExtractFilePath($AFileName) . $Result . $AFileExt;
  }
  else {
    $Result = ExtractFilePath($AFileName) . MidStr($Result, 1, $LPos - 1) . $AFileExt;
  }
  return $Result;
}
// function IncludeLeadingPathDelimiter(const AFileName: string) : string;
function IncludeLeadingPathDelimiter($AFileName) {
  // Result : string;
  $Result = "";
  $Result = $AFileName;
  if (Length($Result) > 0 && !in_array($Result[(1) - 1], AllowDirectorySeparators())) {
    $Result = DirectorySeparator . $Result;
  }
  return $Result;
}
// function IncludeTrailingPathDelimiter(const AFileName: string) : string;
function IncludeTrailingPathDelimiter($AFileName) {
  // Result : string;
  $Result = "";
  $Result = $AFileName;
  if (Length($Result) > 0 && !in_array($Result[(Length($Result)) - 1], AllowDirectorySeparators())) {
    $Result = $Result . DirectorySeparator;
  }
  return $Result;
}
?>

File: tools/pas2php-code-6/pas2php_include/__sysutils.pas
{
 ***************************************************************************
 *                                                                         *
 *   This source is free software; you can redistribute it and/or modify   *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 *   This code is distributed in the hope that it will be useful, but      *
 *   WITHOUT ANY WARRANTY; without even the implied warranty of            *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU     *
 *   General Public License for more details.                              *
 *                                                                         *
 *   A copy of the GNU General Public License is available on the World    *
 *   Wide Web at <http://www.gnu.org/copyleft/gpl.html>. You can also      *
 *   obtain it by writing to the Free Software Foundation,                 *
 *   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.        *
 *                                                                         *
 ***************************************************************************
}

unit __SysUtils;

interface

uses PhpLib, StrUtils, SysUtils;

const

  EmptyStr = '';

  SBoolTrue = 'True';
  SBoolFalse = 'False';

type

  EAbort = class(Exception)
  end;

  EConvertError = class(Exception)
  end;

  EAssertionFailed = class(Exception)
  end;

  EArgumentException = class(Exception)
  end;

  EArgumentOutOfRangeException = class(EArgumentException)
  end;

  EAbstractError = class(Exception)
  end;

  EFinally = class(Exception)
  end;

procedure Abort;
function AnsiCompareStr(const AString1, AString2: string): integer;
function AnsiCompareText(const AString1, AString2: string): integer;
function BoolToStr(const ABoolean: boolean; const ATrue: string = 'True';
  const AFalse: string = 'False'): string;
function ChangeFileExt(const AFileName, AFileExt: string): string;
function DirectoryExists(const AFileName: string): boolean;
function ExtractFileDir(const AFileName: string): string;
function ExtractFileExt(const AFileName: string): string;
function ExtractFileName(const AFileName: string): string;
function ExtractFilePath(const AFileName: string): string;
function FileExists(const AFileName: string): boolean;
function FloatToStr(const AExtended: extended): string;
function GetEnvironmentVariable(const AName: string): string;
function IntToStr(const AInteger: integer): string;
function SameText(const AString1, AString2: string): boolean;
function SetDirSeparators(const AString: string): string;
function StrToBool(const AString: string): boolean;
function StrToBoolDef(const AString: string; const ADefault: boolean): boolean;
function StrToFloat(const AString: string): extended;
function StrToFloatDef(const AString: string; const ADefault: extended): extended;
function StrToInt(const AString: string): integer;
function StrToIntDef(const AString: string; const ADefault: integer): integer;
function Trim(const AString: string): string; external;
function TrimLeft(const AString: string): string;
function TrimRight(const AString: string): string;
function TryStrToBool(const AString: string; out ABoolean: boolean): boolean;
function TryStrToFloat(const AString: string; out AExtended: extended): boolean;
function TryStrToInt(AString: string; out AInteger: integer): boolean;
procedure FreeAndNil(var AVariant: variant);
function IncludeLeadingPathDelimiter(const AFileName: string): string;
function IncludeTrailingPathDelimiter(const AFileName: string): string;

implementation

procedure Abort;
begin
  raise EAbort.Create('Abort');
end;

function IntToStr(const AInteger: integer): string;
begin
  Result := php_strval(AInteger);
end;

function SetDirSeparators(const AString: string): string;
var
  LIndex: integer;
begin
  Result := AString;
  for LIndex := 1 to Length(Result) do begin
    if (Result[LIndex] = '\') or (Result[LIndex] = '/') then begin
      Result[LIndex] := DirectorySeparator;
    end;
  end;
end;

function TryStrToInt(AString: string; out AInteger: integer): boolean;
begin
  if (Length(AString) > 0) and (AString[1] = '$') then begin
    AString := php_base_convert(Copy(AString, 2, MaxInt), 16, 10);
  end;
  AInteger := php_intval(AString, 10);
  Result := php_strval(AInteger) = AString;
end;

function StrToInt(const AString: string): integer;
begin
  if not TryStrToInt(AString, Result) then begin
    raise EConvertError.Create(EmptyStr);
  end;
end;

function StrToIntDef(const AString: string; const ADefault: integer): integer;
begin
  try
    Result := StrToInt(AString);
  except
    Result := ADefault;
  end;
end;

function TryStrToBool(const AString: string; out ABoolean: boolean): boolean;
begin
  try
    if SameText(AString, SBoolTrue) then begin
      ABoolean := True;
    end else if SameText(AString, SBoolFalse) then begin
      ABoolean := False;
    end else begin
      ABoolean := StrToInt(AString) <> 0;
    end;
    Result := True;
  except
    Result := False;
  end;
end;

function StrToBool(const AString: string): boolean;
begin
  if not TryStrToBool(AString, Result) then begin
    raise EConvertError.Create(EmptyStr);
  end;
end;

function StrToBoolDef(const AString: string; const ADefault: boolean): boolean;
begin
  try
    Result := StrToBool(AString);
  except
    Result := ADefault;
  end;
end;

function BoolToStr(const ABoolean: boolean; const ATrue, AFalse: string): string;
begin
  if ABoolean then begin
    Result := ATrue;
  end else begin
    Result := AFalse;
  end;
end;

function FloatToStr(const AExtended: extended): string;
begin
  Result := php_strval(AExtended);
end;

function TryStrToFloat(const AString: string; out AExtended: extended): boolean;
begin
  AExtended := php_floatval(AString);
  Result := php_strval(AExtended) = AString;
end;

function StrToFloat(const AString: string): extended;
begin
  if not TryStrToFloat(AString, Result) then begin
    raise EConvertError.Create(EmptyStr);
  end;
end;

function StrToFloatDef(const AString: string; const ADefault: extended): extended;
begin
  try
    Result := StrToFloat(AString);
  except
    Result := ADefault;
  end;
end;

procedure FreeAndNil(var AVariant: variant);
begin
  AVariant := Null;
end;

function SameText(const AString1, AString2: string): boolean;
begin
  Result := php_strcasecmp(AString1, AString2) = 0;
end;

function TrimLeft(const AString: string): string;
begin
  Result := php_ltrim(AString);
end;

function TrimRight(const AString: string): string;
begin
  Result := php_rtrim(AString);
end;

function AnsiCompareStr(const AString1, AString2: string): integer;
begin
  Result := php_strcmp(AString1, AString2);
end;

function AnsiCompareText(const AString1, AString2: string): integer;
begin
  Result := php_strcasecmp(AString1, AString2);
end;

function GetEnvironmentVariable(const AName: string): string;
begin
  Result := php_getenv(AName);
end;

function ExtractFileDir(const AFileName: string): string;
begin
  Result := php_dirname(SetDirSeparators(AFileName), 1);
end;

function ExtractFileName(const AFileName: string): string;
begin
  Result := php_basename(SetDirSeparators(AFileName));
end;

function FileExists(const AFileName: string): boolean;
begin
  Result := php_is_file(AFileName);
end;

function DirectoryExists(const AFileName: string): boolean;
begin
  Result := php_is_dir(AFileName);
end;

function ExtractFileExt(const AFileName: string): string;
var
  LPos: integer;
begin
  Result := ExtractFilename(AFileName);
  LPos := RPos(ExtensionSeparator, Result);
  if LPos = 0 then begin
    Result := EmptyStr;
  end else begin
    Result := Copy(Result, LPos, MaxInt);
  end;
end;

function ExtractFilePath(const AFileName: string): string;
begin
  Result := ExtractFileDir(AFileName) + DirectorySeparator;
end;

function ChangeFileExt(const AFileName, AFileExt: string): string;
var
  LPos: integer;
begin
  Result := ExtractFilename(AFileName);
  LPos := RPos(ExtensionSeparator, Result);
  if LPos = 0 then begin
    Result := ExtractFilePath(AFileName) + Result + AFileExt;
  end else begin
    Result := ExtractFilePath(AFileName) + Copy(Result, 1, LPos - 1) + AFileExt;
  end;
end;

function IncludeLeadingPathDelimiter(const AFileName: string): string;
begin
  Result := AFileName;
  if (Length(Result) > 0) and not (Result[1] in AllowDirectorySeparators) then begin
    Result := DirectorySeparator + Result;
  end;
end;

function IncludeTrailingPathDelimiter(const AFileName: string): string;
begin
  Result := AFileName;
  if (Length(Result) > 0) and not (Result[Length(Result)] in AllowDirectorySeparators) then begin
    Result := Result + DirectorySeparator;
  end;
end;

end.
